src/main/java/org/checkstyle/autofix/recipe/UpperEll.java

```diff
@@ -43,23 +63,128 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        return new UpperEllVisitor();
+        return new UpperEllVisitor(violations);
     }
 
-    /**
-     * Visitor that replaces lowercase 'l' suffixes in long literals with uppercase 'L'.
-     */
     private static final class UpperEllVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        private static final String LOWERCASE_L = "l";
+        private static final String UPPERCASE_L = "L";
+
+        private final List<CheckstyleViolation> violations;
```

Comment:
Can we reuse this list from the outer class? This will require making `UpperEllVisitor` non-static, but this is ok to avoid such duplication